### ISSUE TITLE
Make it usable for other projects

### ISSUE THREAD

**Issue by thenotsoft** [yiisoft/yii-dev-tool#90]
Now this tool is used only for Yii 3, but it would be convenient to use it for developing other projects, for example, a project based on modules.

### What needs to be changed/added:

- [ ]  Extract core files to `yiisoft/dev-tool` and add dependency it to `yii-dev-tool`

- [ ] Make configuration of repositories possible, decouple from yii

- [ ] Add a command that creates a package from a template

- [x] Add a command to run unit tests in packages

**Comment by samdark** [comment]
> Make configuration of repositories possible, decouple from yii

Isn't it there? https://github.com/yiisoft/yii-dev-tool/blob/master/packages.local.php.example

> Add a command that creates a package from a template

That is easy to do via GitHub currently:

![image](https://user-images.githubusercontent.com/47294/91435754-7e86d180-e86f-11ea-9a34-41c0638e9c61.png)

**Comment by thenotsoft** [comment]
> > Make configuration of repositories possible, decouple from yii
> 
> Isn't it there? https://github.com/yiisoft/yii-dev-tool/blob/master/packages.local.php.example
> 
> > Add a command that creates a package from a template
> 
> That is easy to do via GitHub currently:
> 
> ![image](https://user-images.githubusercontent.com/47294/91435754-7e86d180-e86f-11ea-9a34-41c0638e9c61.png)

1. I'm talking about this https://github.com/yiisoft/yii-dev-tool/blob/master/src/Command/InstallCommand.php#L119 there is still 
    where it is tied.
    Make the default repository address so that you can simply specify `test/test`, not the full path to the repository.

2. It is possible to do with the help of github, but it is not as convenient as `./dev-tool package/create yiisoft/test`
    Make a template folder that you can change if needed for yourself, make automatic insertion of the package name into 
    composer.json, etc.
    And someone uses gitlab or bitbucket, it's easier to specify the repository and then the script will do everything

**Comment by samdark** [comment]
Alright. Makes sense but not a priority at this stage.

**Comment by thenotsoft** [comment]
> Alright. Makes sense but not a priority at this stage.

Of course, this is not a priority now, I created iusse so as not to forget, and so I have already forgotten some ideas)
